inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -884,6 +922,105 @@ impl_types layout_optimizer::get_preferred_impl_type(program_node& node) {
+    } else if (node.is_type<pooling>() && use_onednn_impls) {
+        std::vector<format> onednn_optimized_formats = {
+            format::b_fs_zyx_fsv16,
+            format::b_fs_zyx_fsv32,
+            format::b_fs_yx_fsv32,
+            format::b_fs_yx_fsv16,
+            format::bs_fs_yx_bsv16_fsv16,
+            format::bs_fs_zyx_bsv16_fsv16,
+            format::bs_fs_yx_bsv32_fsv16,
+            format::bs_fs_yx_bsv32_fsv32,
+            format::bs_fs_yx_bsv4_fsv4,
+            format::bs_fs_yx_bsv4_fsv2,
+            // format::bs_fs_zyx_bsv4_fsv4,
+            // format::bs_fs_zyx_bsv4_fsv2,
+        };
+
+        impl_types impl_candidate = impl_types::onednn;
+
+        // Unexpected layout
+        if (std::find(onednn_optimized_formats.begin(), onednn_optimized_formats.end(), preferred_format) == onednn_optimized_formats.end()) {
+            impl_candidate = impl_types::ocl;
+        }
+
+        // if (node.is_type<convolution>()) {
+        //     // onednn doesn't have good support for groups with fsv16 fmt
+        //     auto& conv = node.as<convolution>();
+        //     auto input_layout = conv.input().get_output_layout();
+        //     bool fp16_input = input_layout.data_type == data_types::f16;
+        //     bool has_groups = conv.get_primitive()->groups > 1;
+        //     bool is_depthwise = conv.get_primitive()->groups == input_layout.size.feature[0];
+        //     bool first_conv = input_layout.size.feature[0] <= 4;
+        //     bool enable_onednn_dw_fp16_conv = fp16_input && is_depthwise;
+        //     if (((has_groups && !enable_onednn_dw_fp16_conv) || first_conv) &&
+        //         (conv.get_output_layout().format == format::b_fs_yx_fsv16)) {
+        //         impl_candidate = impl_types::ocl;
+        //     }
+        // }
+
+        // if (node.is_type<deconvolution>()) {
+        //     auto& deconv = node.as<deconvolution>();
+        //     auto input_layout = deconv.input().get_output_layout();
+        //     bool valid_ic = input_layout.size.feature[0] >= 16;
+        //     bool valid_groups = deconv.get_primitive()->groups == 1;
+        //     bool valid_post_ops = get_post_ops_count(node) <= 10;
+        //     bool valid_batch = input_layout.size.batch[0] < 16;  // oneDNNs optimized kernel doesn't support big batches yet
+        //     bool valid_params = valid_ic && valid_groups && valid_post_ops && valid_batch;
+        //     if (!valid_params)
+        //         impl_candidate = impl_types::ocl;
+        // }
+
+        // // [WA] onednn doesn't support > 32 post-ops. Remove once onednn improve post-ops for GPU.
+        // if (get_post_ops_count(node) > 32) {
+        //    impl_candidate = impl_types::ocl;
+        // }
+
+        // if (!data_types_are_suitable_for_onednn(node)) {
+        //     impl_candidate = impl_types::ocl;
+        // }
+
+        // for (auto& fo : node.get_fused_primitives()) {
```

Comment:
Why do you commented out these changes?

Reply:
This code is a dead code at this moment because we did not integrated fusing, yet. So it was just left as comments.